airbyte-integrations/connectors/source-close-com/source_close_com/source.py

```diff
@@ -0,0 +1,479 @@
+        return params
+
+
+class AnsweredDetachedCallTasks(IncrementalCloseComStream):
+    cursor_field = "date_created"
+    _type = "answered_detached_call"
+
+    def path(self, **kwargs) -> str:
+        return "task"
+
+    def request_params(self, stream_state=None, **kwargs):
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        params["_type"] = self._type
+        if stream_state.get(self.cursor_field):
+            params["date_created__gt"] = stream_state.get(self.cursor_field)
+        return params
+
+
+class VoicemailTasks(IncrementalCloseComStream):
+    cursor_field = "date_created"
+    _type = "voicemail"
+
+    def path(self, **kwargs) -> str:
+        return "task"
+
+    def request_params(self, stream_state=None, **kwargs):
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        params["_type"] = self._type
+        if stream_state.get(self.cursor_field):
+            params["date_created__gt"] = stream_state.get(self.cursor_field)
+        return params
+
+
+class OpportunityDueTasks(IncrementalCloseComStream):
+    cursor_field = "date_created"
+    _type = "opportunity_due"
+
+    def path(self, **kwargs) -> str:
+        return "task"
+
+    def request_params(self, stream_state=None, **kwargs):
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        params["_type"] = self._type
+        if stream_state.get(self.cursor_field):
+            params["date_created__gt"] = stream_state.get(self.cursor_field)
+        return params
+
+
+class IncomingSmsTasks(IncrementalCloseComStream):
+    cursor_field = "date_created"
+    _type = "incoming_sms"
+
+    def path(self, **kwargs) -> str:
+        return "task"
+
+    def request_params(self, stream_state=None, **kwargs):
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        params["_type"] = self._type
+        if stream_state.get(self.cursor_field):
+            params["date_created__gt"] = stream_state.get(self.cursor_field)
+        return params
+
+
+class LeadCustomFields(CloseComStream):
+    def path(self, **kwargs) -> str:
+        return "custom_field/lead"
+
+
+class ContactCustomFields(CloseComStream):
+    def path(self, **kwargs) -> str:
+        return "custom_field/contact"
+
+
+class OpportunityCustomFields(CloseComStream):
+    def path(self, **kwargs) -> str:
+        return "custom_field/opportunity"
+
+
+class ActivityCustomFields(CloseComStream):
+    def path(self, **kwargs) -> str:
+        return "custom_field/activity"
+
+
+class Users(CloseComStream):
+    def path(self, **kwargs) -> str:
+        return "user"
+
+
+class SourceCloseCom(AbstractSource):
+    @staticmethod
+    def _convert_auth_to_token(username: str, password: str) -> str:
```

Comment:
Why do we need this method? Do we use Username and Password or just API key for authentication?

Reply:
Change it to Base64HttpAuthenticator that inherits TokenAuthenticator class. TokenAuthenticator doesn't support authentication with username and password.